### ISSUE TITLE
Angular Implementation

### ISSUE THREAD

**Issue by ericrovtar** [honeybadger-io/honeybadger-webpack#6]
I love that you're supporting Webpack, but (unless I'm wrong), some frameworks - like Angular - use Webpack but don't give users direct access to `webpack-config.js`.

Is there a workaround for this or a chance we might see an Angular module released?

**Comment by joshuap** [comment]
Hey @ericrovtar, it *seems* like Angular provides a way to add Webpack plugins:

https://angular.io/guide/webpack#plugins

I am not super familiar with this, so I could be wrong. Does that point you in the right direction at all?

We don't currently have plans to officially build additional Webpack integrations (but we welcome community projects). We *are* currently working on framework documentation for popular JS frameworks -- including Angular -- as a first step towards a more seamless experience. Phase two of that may include developing additional integrations (or at least providing documentation on how to integrate Honeybadger with your framework manually).

**Comment by ericrovtar** [comment]
@joshuap Interesting. All my googling and that never came up. I'll take a look. We haven't done that before, but we really would love automating the source map uploads. Thanks!

**Comment by joshuap** [comment]
No problem! If you do find a configuration that works for Angular, would you mind commenting back here with an example? I'll add it to our documentation.

**Comment by ericrovtar** [comment]
@joshuap This might not be the best place to follow up, so let me know if there's a better medium.

I think I have a solution. I got the main source map to upload (ex: `main.0eb484898c98a0719f5c.bundle.js`), but an Angular build will usually have multiple bundles:

- inline.*.bundle.js
- main.*.bundle.js
- polyfills.*.bundle.js
- scripts.*.bundle.js

Each of these will have its own source map.

I know there's the `<additional source files>` option when using your API, so I can tell it of the additional source files, but how do I tell Honeybadger about the additional source maps?

For now, I'm going to stick with `main` as the goal. It's the most important anyway.

Thanks!

**Comment by joshuap** [comment]
@ericrovtar I don't believe this plugin currently supports uploading multiple source maps; we'll have to review if it's something we want to add. Can you confirm, @monde?

In the meantime, if `main` contains most of your application code, I think it should still be pretty useful. If you have any account-specific questions feel free to email us at support@honeybadger.io.

**Comment by monde** [comment]
I have some ideas regarding multiple source maps @ericrovtar, I will try to address them in the next couple of days

**Comment by ericrovtar** [comment]
Thanks, @joshuap & @monde!

**Comment by joshuap** [comment]
@monde when you get the chance, could you add any ideas you had here? Thanks!

**Comment by monde** [comment]
(I'm not making a promise!) but I'll try and hack on an example to find out more tonight.

**Comment by monde** [comment]
FYI the angular webpack guide has an example app https://angular.io/guide/webpack look for "You can also download the final result.". I fiddled with it a little bit, but the app doesn't seem to be loading but I can see that webpack ran and generated js and map files. The example has a `webpack.config.js` that points to `config/webpack.dev.js` which has a plugin stanza to hook into. @ericrovtar let me know if this enough to get you along otherwise I can spend some more time trying to drill out the full example in the next week.

**Comment by ericrovtar** [comment]
@monde I appreciate you continuing to look into this. For now, we're scripted the uploading the main JS map to Honeybadger during our build and deploy process. It's working for us. When I have the chance, I'll take a look at what you found, though. Thanks!
